### ISSUE TITLE
Manual Backup / Export option

### ISSUE THREAD

**Issue by 8bitPit** [8bitPit/Niagara-Issues#222]
See: https://help.niagaralauncher.app/article/37-backup-settings

Edit: 
Implemented in v1.13.0 🎉 
For more information, please refer to [Backup & Restore](https://help.niagaralauncher.app/article/179-backup-restore)

**Comment by cauerego** [comment]
please do it automated!

**Comment by billouetaudrey** [comment]
Please add this feature !!! I'm a testers for Redmi Note 4 so changed rom really often ( 4/5 per day ) and always need to hide all my unwant app (Swift Backup doesn't recognize this backup ... )
Thanks

**Comment by Desertplanet** [comment]
Please add this feature.. Pain to have to redo look etc everytome I reset my phone to clean things up..
Thanks

**Comment by Montell2099** [comment]
This feature would be a godsent imo, being able to backup different customized icon configurations etc.

**Comment by billouetaudrey** [comment]
Hello, please add this features, maybe if I used extra app like sb or tbp, Pro features isn't restore ( and can't be restore )

**Comment by 8bitPit** [comment]
@billouetaudrey Google does manage in-app-purchases and stores them separately (they're bound to your Google Play account). They're not stored in the app itself. Are you using the same Google Play accounts for both devices?

**Comment by billouetaudrey** [comment]
> @billouetaudrey Google does manage in-app-purchases and stores them separately (they're bound to your Google Play account). They're not stored in the app itself. Are you using the same Google Play accounts for both devices?

I changed ROM and when I restore Niagara, I need to clear data and do setting for have pro back 😉

**Comment by 8bitPit** [comment]
Then Google should automatically "restore" your purchase then.

**Comment by billouetaudrey** [comment]
> Then Google should automatically "restore" your purchase then.

I will retry when changing ROM again 😃

**Comment by Montell2099** [comment]
I would _love_ to be able to manually save the current icon configuration! But it could perhaps also be handy being able to manually backup the entire Niagara configuration.

**Comment by eche779** [comment]
Yeah I would like a complete manual backup, Tu easily switch between favourite setups back and forth

**Comment by Desertplanet** [comment]
Please add also... 😊

Often I reset phone for clean fresh install.. And would be easier to load a saved config.. From the cloud if possible.. 
Thanks

**Comment by dark-N00B** [comment]
Export option would be great to switch between multiple restore points for more than one device or so!

**Comment by 8bitPit** [comment]
Thanks for your feedback :D

**Comment by Montell2099** [comment]
> Thanks for your feedback :D

Is this currently a low-priority task? I'm playing around with new icons at the moment, and are being reminded of how convenient it would be to save a configuration — so that I then can experiment crazy with a safe option to revert to the previous setup.

**Comment by 8bitPit** [comment]
@Montell2099 it is, because we're working on something else that would also help you a lot in your scenario :)

**Comment by Ultimanubis** [comment]
Manual backups would be nice.

**Comment by CerberusCode** [comment]
Niagara should have this features, most of a launcher have this backup/restore. 👍

**Comment by Montell2099** [comment]
> Niagara should have this features, most of a launcher have this backup/restore. 👍

Yes, indeed. I figure it's a complex system & not something that can be made in a heartbeat. But I hope something will be put in place at some time.

> @Montell2099 it is, because we're working on something else that would also help you a lot in your scenario :)

The developer has even stated that something helpful will be made available sometime. I'm definitely keeping my hopes up, but not holding my breath.

**Comment by yuping** [comment]
Please have manual backup/restore options, that could help user have different setting per device.
e.g. my android phone and my tablet has different setting I need different backup and restore option.

**Comment by gdsingh94** [comment]
Please add the manual backup/restore

**Comment by lukas-rohrer** [comment]
manual backup/restore would be awesome for people like me with a degoogled device

**Comment by Montell2099** [comment]
Still keeping my hope up in mid-2021 for this feature. 🤞🏼

**Comment by yuping** [comment]
Indeed. that is MUST have feature for those people who has multiple devices.

**Comment by Montell2099** [comment]
> Indeed. that is MUST have feature for those people who has multiple devices.

True indeed! It's also a very powerful tool for dedicated power-users—who are keen on changing their look every now and then, fiddle with different widgets etc—to have multiple backup-files ready to load in case of you want to do a "roll back" to a specific date with a specific setup of icons, widgets, Niagara settings etc.

**Comment by nr431** [comment]
I would love an export all config option, especially if it included, icons, widgets,and folders.it takes time to set these up,and the ability to use the saved configuration as a starting point on other devices would be a huge plus.

**Comment by Montell2099** [comment]
> @Montell2099 it is, because we're working on something else that would also help you a lot in your scenario :)

Could you perhaps tease what this other solution might be?

I am very enthusiastic about testing out different home screen setups within Niagara. But when I'm feeling contempt with my current setup, I simply do not dare to start mixing things around—just because of how time consuming it would be to restore each icon separately—especially since I mix icons from different icon packs.

Cheers!

**Comment by Kamilo225** [comment]
I would love to be able to manually save my current settings such as favorites list, folders and icon pack

**Comment by Montell2099** [comment]
@8bitPit Is this a possible feature in the foreseeable future? Would love to make manual backups, or just backups of _some_ sort.

**Comment by leighfwarren** [comment]
@8bitPit This would be a great feature, shame it's been on the list for so long with no action.

**Comment by entrealist** [comment]
+100 for manual backup

**Comment by IQHaitham89** [comment]
Manual backups would be nice⭐💙

**Comment by Gittegatt** [comment]
+1 for implemention of manual backups export/import. Please! 🙏

**Comment by cjjadeja** [comment]
Manual backups are a "must have". There are endless benefits to it.

**Comment by ruslanabdulla** [comment]
+100! for manual backup :)

**Comment by rcdailey** [comment]
I just restored my backup from a OnePlus 7 Pro to a Pixel 6 Pro using USB cable and none of my settings got carried over...

**Comment by devz3r0** [comment]
A backup to a local choosable directory, as I am not using google cloud.

**Comment by Gittegatt** [comment]
> A backup to a local choosable directory, as I am not using google cloud.

If I am right, I mean Google does not storing application data in the backup itself.

**Comment by devz3r0** [comment]
> > A backup to a local choosable directory, as I am not using google cloud.
> 
> If I am right, I mean Google does not storing application data in the backup itself. Just the information which apps are installed as a 'list'.

for what is it then? Then you don't need to do a backup at all if you just need to reenter the data by yourself again.

**Comment by dgustala** [comment]
Please please implement this feature asap. It is awful to change my mobile and lose all my configs!!

**Comment by ithorien** [comment]
I would also echo this; restored, and at first all my faves/folders showed. Then as apps were reinstalling, Niagra wiped everything and since it's "automatic", I now have no way of bringing it back. Terrible oversight.

**Comment by soymadip** [comment]
Although it's labelled as low priority, I guess it's an very important and basic feature to able to backup & restore  launcher's configuration.

**Comment by Geotelea** [comment]
Please do it ...it is bad every time to begin again and again .

**Comment by Hydrometry** [comment]
Google's backup solution simply does not work for me. Seems like a manual solution is necessary for the non-rooted folks, at least until Google makes their solution more reliable. Nova has it.

Does Swift Backup actually work reliably with Niagara for anyone? Seems like the backup works only half the time, when it's > 1 MB in size.

**Comment by algy21** [comment]
May as well throw my +1 in there. So time consuming customizing Niagara and no way to backup the configuration. It's a huge disappointment that this isn't done, because Niagara is without a doubt the best Launcher out there IMHO and I've tried most. Microsofts Launcher lasted longest but it's problems with icons drove me to find something else and there was Niagara, new on the market and so logical. However it needs backup, please and thank you

**Comment by Msouza91** [comment]
The backup feature never worked for me, not once... I am a pro user, but had to stop using the launcher because every time I needed to change phones or format my current device I'd lose all my folders and other configurations. And I tried manually start a backup on Google one app, restored from cloud, cable, the whole nine yards and nothing... It already keeps track of everything went not spir a JSON or yaml file with the configurations as well as the automated backup as a safety net? It is beyond me the 4 years later and this still is an issue...

**Comment by ciskonc** [comment]
Please! Make Manual Backup Export option

**Comment by skoshy** [comment]
I just lost my whole home screen because I didn't have a backup. I turned on Samsung's extreme battery saver mode; when turning it off and seeing Niagara again, all my shortcuts are gone, even after a reboot.

For the reason of no manual backups alone, I am not using Niagara again.

**Comment by tjveith** [comment]
Please DEFINITELY add the manual back up / export option!!

**Comment by resiehnnes** [comment]
I think manual backup of configuration file should always come first as it is most reliable and safe (private) method than any cloud backup, also it would be handy for those who do not have google apps and services in their phone.

+1 for manual backup exporting/importing configuration file

**Comment by Jamiermusic** [comment]
It's the feature which keeps me from committing fully and purchase pro. 

I'd love to see a manual backup similar to Nova launcher. 

Throwing in there, the ability to have more than 1 widget on home screen. I like to see Todo and calendar on home screen 😊 besides this, love the launcher!

**Comment by brandonwardlaw** [comment]
Definitely needs a manual backup option. Very surprised that this isn't a feature given how common it is in the Android app ecosystem. Also quite frustrated that the current backup feature doesn't seem to work when migrating between Samsung devices via SmartSwitch.

**Comment by NefariousFerris** [comment]
please add this, I invested so much time setting up my homescreen, popups, and widgets just right and if I lose any of it I would just give up on using the launcher again

**Comment by bbecausereasonss** [comment]
How is this not a feature? The automatic backup does not work! Just got a pixel phone and have to redo everything? No thanks. Not using this launcher anymore.. Good grief this should have been default!

**Comment by topsy-turvy99** [comment]
Proper implementation that works on all devices properly without any issues takes time :), There already exists auto backup using Android default backup system but it doesn't work at times, but tbh  Setting up Niagara is much faster compared to any other launchers, personally i never felt the need of it but anyways Devs are working on it, and you'll hopefully get to see native backup feature anytime soon :), just a little patience.


Meanwhile you can contribute by filling out the form:
https://github.com/8bitPit/Niagara-Issues/issues/222#issuecomment-1176191860

**Comment by schlice** [comment]
+1 for manual backup and restore.  Also experienced automatic restore not working, even when using cable transfer.

**Comment by clunkiersalt817** [comment]
Yes manual backup is a must have feature

**Comment by pruchkin2** [comment]
Yes manual backup is a must have feature

**Comment by ruslanabdulla** [comment]
+100 for manual backup and restore!

**Comment by daedgoco** [comment]
I see it is now planned, can't wait to use it.

**Comment by Msouza91** [comment]
> Proper implementation that works on all devices properly without any issues takes time :), There already exists auto backup using Android default backup system but it doesn't work at times, but tbh Setting up Niagara is much faster compared to any other launchers, personally i never felt the need of it but anyways Devs are working on it, and you'll hopefully get to see native backup feature anytime soon :), just a little patience.
> 
> Meanwhile you can contribute by filling out the form: [#222 (comment)](https://github.com/8bitPit/Niagara-Issues/issues/222#issuecomment-1176191860)

Not sure how this feature has device specific complexity, the settings are most likely saved to a file in a format specif to Niagara.

None of the settings are brand or model bound, it just needs to save this internal file, be it a json, yaml or encripted binary to a folder chosen by the user, as I see it is even easier than the "automagic" backup that being completely candid has worked a total of 0 times in 1,5 to 2 years I have tried using Niagara; more reliable as well and would be better aligned with the privacy and user control ideals of the launcher.

Thankfully, it is being worked on now so at least we don't have to have this discussion anymore. :)

**Comment by Onomic** [comment]
Yes really need this as there is no way on Samsung phones to restore. They removed google restore and force smart switch which doesn't work with Niagara. Need a way to restore on a new device.

**Comment by blogmangit** [comment]
+1 from me really need this niagara pro user for lifetime.

**Comment by thejaustin** [comment]
Any updates?

**Comment by EyesofBucket** [comment]
+1 Hoping for a plaintext format like yaml or json.  User readability and editability would be huge as well.

**Comment by QuentinSc** [comment]
Can't buy the app without this feature

**Comment by affan2** [comment]
+1 for any backup method that actually works!

**Comment by murielg** [comment]
+1. Pro user. Just set up a new phone and the auto back-up did not back up any of my Niagara settings 😢

**Comment by murielg** [comment]
@blogmangit https://help.niagaralauncher.app/article/37-backup-settings apparently? but it doesn't seem to have worked for me. I am a newer Niagara user so this is the first time I've switched phones and none of the settings seem to have been backed up 🤷

**Comment by dgustala** [comment]
![Screenshot_20231022_202926_Google One](https://github.com/8bitPit/Niagara-Issues/assets/5603907/f04151ae-df85-4105-bf15-9c0d5c18b430)
Do we need other proof?

**Comment by soymadip** [comment]
> ![Screenshot_20231022_202926_Google One](https://user-images.githubusercontent.com/5603907/277199430-f04151ae-df85-4105-bf15-9c0d5c18b430.jpg) Do we need other proof?

Looks like Niagara devs takes cleanliness a bit too much that they wanna keep backend code clean too by not implimenting necessary features too🤔

**Comment by soymadip** [comment]
> Yup, that's what I planned.

I didn't realise this feature is taking this long😯
Looks like we are gonna get very special backup option that no other app has untill now😌

**Comment by NelsonMinar** [comment]
Just migrated from a Pixel 6 to Pixel 8 (via wireless backup / restore) and while many things transferred over, Niagara didn't. I got a message during the new phone setup process that "Niagara Launcher keeps quitting" and clicked the "stop process" button to let the phone setup go. Now it looks like I'll probably have to manually reconfigure Niagara Launcher on the new phone. I'm not mad about this but I agree some sort of app-controlled settings process would be a big help. I've got a particular challenge in that I run Niagara on both a phone and a tablet with very different setups, a single config for both would not work.

**Comment by sharkboy777** [comment]
How long is it gonna take for a simple backup feature to be implemented? You guys have clearly mentioned in your help section that it's coming and it's been 4 years since then.

What is the use of buying a PRO version of I don't get a simple backup option. Seems like the Google backup is not working properly at all.

This app is not useful for me if there is no backup option.

**Comment by Lorestan00** [comment]
This really should be a priority...

**Comment by tjveith** [comment]
I agree with everyone. I posted a recommendation several years ago for a backup. It's ridiculous that all of his users are saying they want to back up yet he doesn't care. Everyone is paid for this app, where is the commitment from the developer. A complete joke. If I didn't like this app so much, I wouldn't use it because of the developers attitude, make a backup for God's sakes, seriously.

**Comment by blogmangit** [comment]
chill down guys this year saw a lot of development, but backup is a delicate one feature. most of the users will ask questions regarding collecting information and permissions, etc.

**Comment by tjveith** [comment]
I'm sure everyone appreciates all of the development. But the backup system has been requested year after year after year. Everyone is asking for it, it is one of the most requested features. It makes going from one phone to the next much easier. I don't think you have a right to tell people to chill out when the developer is clearly not listening to customer feedback. People paid for this application, most of which had no idea there was no backup in place when they first purchased it. Even Nova has a backup system. Relying on the Google backup system is archaic for this type of app. People are understandably upset because they have been asking for this feature for years. Developers need to listen to their customers a little more closely, that is what makes successful software.

**Comment by blogmangit** [comment]
> I'm sure everyone appreciates all of the development. But the backup system has been requested year after year after year. Everyone is asking for it, it is one of the most requested features. It makes going from one phone to the next much easier. I don't think you have a right to tell people to chill out when the developer is clearly not listening to customer feedback. People paid for this application, most of which had no idea there was no backup in place when they first purchased it. Even Nova has a backup system. Relying on the Google backup system is archaic for this type of app. People are understandably upset because they have been asking for this feature for years. Developers need to listen to their customers a little more closely, that is what makes successful software.

let's say backup is implemented then what's next ?

**Comment by Lorestan00** [comment]
This must be easily one of the most commented on areas for development. As pointed out what makes this frustrating is that it was raised 5 years ago and it was benign looked into then. 

It's also not a niche feature but one several major launchers have had for years. It's a pinpoint for moving to new phones but also if you want to backup a few different setups with certain icon pack with customise icons. 

Data collection isn't really an issue as there can (and should be) an option to backup locally. 

Let's get this one out of the way before we starting looking ahead!

**Comment by psychowood** [comment]
> chill down guys this year saw a lot of development, but backup is a delicate one feature. most of the users will ask questions regarding collecting information and permissions, etc.

While I agree that it must be done with a grain of salt, I don't see why it should need more permissions, the export can be done quote easily. The hard part is restoring since it needs to keep a lot of things in check, e.g. linking an app that is not installed, handling different screen resolution,  accessing missing widgets, different icons pack, and so on.

**Comment by evetesla** [comment]
+1

**Comment by hillihappo** [comment]
I don't know if someone already mentioned this but when I changed phone Niagara didn't restored my backup but then I uninstalled the app and reinstalled it the backup restored automatically without doing anything. So try that if you have switched phone but the initial backup haven't been successful.

**Comment by biikman** [comment]
Just started using the 7 day trial with the intent of buying it, but unfortunately no baked in backup option is a deal breaker. It's to bad, as I really do appreciate this launcher. I'll definitely keep tabs on it and reconsider once it's available.

**Comment by biikman** [comment]
Apologies if I broke a rule in my previous comment that seems to have been removed? Just commented on disappointment on no backup option, and would purchase if it had it. If this also gets removed please advise on reason.

**Comment by Lorestan00** [comment]
You're right to be disappointed. It's been raised for years. Summer feature drop is completely meaningless. Hard to believe comments and suggestions in the forum keep falling on deaf ears

**Comment by trumad** [comment]
Yeah, I didn't really need new icons. I'm happy with the default ones. Backup/restore is a much needed feature, given the number of websites and special shortcuts I've added to my app list and pop-ups.

**Comment by Lorestan00** [comment]
Same.  I genuinely get worried I'm going to lose my layout. Lots of folders and shortcuts.

**Comment by anthonyma94** [comment]
This feature has been requested for 4 years. What will it take for the dev to notice? I am normally a reasonable guy, but a basic feature "in development" for 4 years while you push out arguably useless features? Unacceptable. @8bitPit what do you have to say for yourself?

**Comment by ConL643** [comment]
I want to try the new fonts that they have talked about for the last 6 months, but I won't do it without a backup. I like my current setup, but I am really worried I will lose it and have to go through the whole process again.

**Comment by trankillity** [comment]
Prepare for a new wave of disgruntled users as people the world over start receiving their Pixel 9s...

Yes, I'm one of them. So stupid that this hasn't been fixed yet.

**Comment by mdcollins05** [comment]
I ran into this issue today. I swapped phones and copied data over from my old to my new phone. Unfortunately, while some setting seemingly copied over, I still had to re-add all my favorites and pop up folders. I would love a manual backup solution to avoid this in the future.

**Comment by Lorestan00** [comment]
It won't happen. This thread is 6 years old if they were going to do it they would have by now. I don't understand the reluctance to put in such a basic feature which is so heavily demanded. 

Can one of the mods explain this ?

**Comment by yuping917** [comment]
I can't understand why it still hasn't happened either, at least making it one of the pro feature will be good though.

**Comment by anthonyma94** [comment]
We need to review bomb it on the Play store until the dev listens, he clearly doesn't pay attention here.

**Comment by dipo70** [comment]
Ich glaube der Entwickler kann es nicht, oder hat keine Lust dazu. Stattdessen werden wir mit nutzlosen Verschönerungen abgelenkt.

Ich nutze 5 Android Geräte gleichzeitig und jede konfiguriert des Launchers sieht anders aus. Das nervt ungemein.

**Comment by Nasyncs** [comment]
The way they ask for $30 for this and a simple feature like this hasn't ben added speaks volumes. You'd expect the rev to be more responsive and add features to make people feel like they got their money's worth. Apparently one of the justifications are updated but nothing new has been added in the last year...

**Comment by DovieW** [comment]
Peter Huber talks about it [here](https://www.androidpolice.com/niagara-launcher-creator-peter-huber-interview/?taid=66f344b1c22cdb00016f6113&utm_campaign=trueanthem&utm_medium=social&utm_source=twitter)

> Peter told me that Niagara users have been asking for a backup and restore feature for their launcher settings because it would simplify migration to a new device and also save time in case of a botched update.
> He promised this option is also on the road map. "We have a basic version running internally, but some things aren't so trivial," he said, referring to critical issues slowing down development. "I think with Android 14 it was, Google completely restricted access to the system wallpaper, and you can't back up the wallpaper unless the user applies it through Niagara's wallpaper picker."

**Comment by Nasyncs** [comment]
> Peter Huber talks about it [here](https://www.androidpolice.com/niagara-launcher-creator-peter-huber-interview/?taid=66f344b1c22cdb00016f6113&utm_campaign=trueanthem&utm_medium=social&utm_source=twitter)
> 
> > Peter told me that Niagara users have been asking for a backup and restore feature for their launcher settings because it would simplify migration to a new device and also save time in case of a botched update.
> > He promised this option is also on the road map. "We have a basic version running internally, but some things aren't so trivial," he said, referring to critical issues slowing down development. "I think with Android 14 it was, Google completely restricted access to the system wallpaper, and you can't back up the wallpaper unless the user applies it through Niagara's wallpaper picker."

No other launcher back up the wallpaper, why is this even attempted? Just back up settings with app/widgets/popul folders.

**Comment by DovieW** [comment]
I agree. It's such an important feature, I think it should be released in any state and it should be improved on as time goes on. I understand they want it to be polished and it to backup everything, but it sounds like they can do the backup on the time consuming stuff.

**Comment by Lorestan00** [comment]
Agreed even a basic implementation on day one with features added would be great

**Comment by kriswilk** [comment]
In the [article](https://www.androidpolice.com/niagara-launcher-creator-peter-huber-interview/) referenced by @Nasyncs, the app author states:

> I think with Android 14 it was, Google completely restricted access to the system wallpaper, and you can't back up the wallpaper unless the user applies it through Niagara's wallpaper picker.

and:

> ...we are in the transition phase right now, from a student project to something bigger, a proper app...

Difficulty backing up wallpaper should not be a feature blocker...we'd be fine without that. It's the tedious customizations that I want to be able to backup.

The reality is that this is a small team that's -- by its leader's own admission -- still kind of treating it as a "student project". Hopefully with some direction and focus, they can bring together the feature we're looking for.

**Comment by Nasyncs** [comment]
> In the [article](https://www.androidpolice.com/niagara-launcher-creator-peter-huber-interview/) referenced by @Nasyncs, the app author states:
> 
> > I think with Android 14 it was, Google completely restricted access to the system wallpaper, and you can't back up the wallpaper unless the user applies it through Niagara's wallpaper picker.
> 
> and:
> 
> > ...we are in the transition phase right now, from a student project to something bigger, a proper app...
> 
> Difficulty backing up wallpaper should not be a feature blocker...we'd be fine without that. It's the tedious customizations that I want to be able to backup.
> 
> The reality is that this is a small team that's -- by its leader's own admission -- still kind of treating it as a "student project". Hopefully with some direction and focus, they can bring together the feature we're looking for.

Which is all fine... But then charging $30 with half baked features for it is not.

**Comment by bbecausereasonss** [comment]
Is this still not a feature? Good lord.

**Comment by evetesla** [comment]
> Is this still not a feature? Good lord. I've had 3 phones already since buying this, and still have to restart over again from scratch every time?

Unfortunately yes 😔

**Comment by biikman** [comment]
Any year now I'm sure..

**Comment by Schoggiweggli** [comment]
Ridiculous that this is still not a thing. Best launcher for me but man that's disappointing.

**Comment by HorstiG** [comment]
$30 without a backup function is a bit bold. Especially if you don't want to root your phone.

**Comment by richardfrost** [comment]
> $30 without a backup function is a bit bold. Especially if you don't want to root your phone.

Can't overstate this. I used the free version about a year ago, and when switched phones it lost it all. I've had a few times where I was thinking of purchasing, but the high price coupled with no reliable backup system is a deal breaker for me. I can't understand why this hasn't had traction after so many years. Oh well, back to my Nova I go.

**Comment by trankillity** [comment]
[About damn time](https://www.youtube.com/watch?v=YEHQlWqzw2c&t=298s). Only a whopping 6 years after this issue was created.

**Comment by TFackler** [comment]
Great news for everyone who has commented on this issue.

We have added a manual backup feature with version 1.13.0, available via our communities already ([Telegram](https://t.me/s/niagara_news), [Discord](https://discord.com/invite/niagara-launcher-728630549278883950)), and will be rolled out on Google Play a couple of weeks later.

It is still being actively worked on and has some limitations. For details, please refer to [Backup & Restore](https://help.niagaralauncher.app/article/179-backup-restore).

We will close this issue and further requests and suggestions should be added to #3068 .
